gcloud-java-examples/src/main/java/com/google/gcloud/examples/dns/DnsExample.java

```diff
@@ -0,0 +1,528 @@
+          System.out.printf("%nName: %s%n", zone.name());
+          System.out.printf("ID: %s%n", zone.id());
+          System.out.printf("Description: %s%n", zone.description());
+          System.out.printf("Created: %s%n", formatter.format(new Date(zone.creationTimeMillis())));
+          System.out.printf("Name servers: %s%n", Joiner.on(", ").join(zone.nameServers()));
+        }
+      } else {
+        System.out.println("Project contains no zones.");
+      }
+    }
+
+    @Override
+    public String params() {
+      return "";
+    }
+
+    @Override
+    public boolean check(String... args) {
+      return args.length == 0;
+    }
+  }
+
+  private static class GetZoneAction implements DnsAction {
+
+    /**
+     * Gets details about a zone with the given name.
+     */
+    @Override
+    public void run(Dns dns, String... args) {
+      String zoneName = args[0];
+      Zone zone = dns.getZone(zoneName);
+      if (zone == null) {
+        System.out.printf("No zone with name '%s' exists.%n", zoneName);
+      } else {
+        System.out.printf("Name: %s%n", zone.name());
+        System.out.printf("ID: %s%n", zone.id());
+        System.out.printf("Description: %s%n", zone.description());
+        DateFormat formatter = new SimpleDateFormat("YYYY-MM-dd HH:mm:ss");
+        System.out.printf("Created: %s%n", formatter.format(new Date(zone.creationTimeMillis())));
+        System.out.printf("Name servers: %s%n", Joiner.on(", ").join(zone.nameServers()));
+      }
+    }
+
+    @Override
+    public String params() {
+      return "<zone_name>";
+    }
+
+    @Override
+    public boolean check(String... args) {
+      return args.length == 1;
+    }
+  }
+
+  private static class DeleteZoneAction implements DnsAction {
+
+    /**
+     * Deletes a zone with the given name.
+     */
+    @Override
+    public void run(Dns dns, String... args) {
+      String zoneName = args[0];
+      boolean deleted = dns.delete(zoneName);
+      if (deleted) {
+        System.out.printf("Zone %s was deleted.%n", zoneName);
+      } else {
+        System.out.printf("Zone %s was NOT deleted. It does not exist.%n", zoneName);
+      }
+    }
+
+    @Override
+    public String params() {
+      return "<zone_name>";
+    }
+
+    @Override
+    public boolean check(String... args) {
+      return args.length == 1;
+    }
+
+  }
+
+  private static class DeleteDnsRecordAction implements DnsAction {
+
+    /**
+     * Deletes a DNS record of type A from the given zone. The last parameter is ttl and it is not
+     * required. If ttl is not provided, a default value of 0 is used. The service requires a
+     * precise match (including ttl) for deleting a record.
+     */
+    @Override
+    public void run(Dns dns, String... args) {
+      String zoneName = args[0];
+      String recordName = args[1];
+      String ip = args[2];
+      int ttl = 0;
+      if (args.length > 3) {
+        ttl = Integer.valueOf(args[3]);
```

Comment:
replace "Integer.valueOf" with "Integer.parseInt" globally (should be 4 occurences)